api/src/db-connection.ts

```diff
@@ -114,6 +114,31 @@ export class Connection {
     return { rows, offset, size, total };
   }
 
+  async getPaperEntityCount(paperSelector: PaperSelector): Promise<{paper: PaperSelector, count: number} | null> {
+    const idField = isS2Selector(paperSelector) ? 'p.s2_id' : 'p.arxiv_id';
+    const idValue = isS2Selector(paperSelector) ? paperSelector.s2_id : `${paperSelector.arxiv_id}%`;
+    const whereClause = `${idField} ${isS2Selector(paperSelector) ? '=' : 'ilike'} ?`
+    const response = await this._knex.raw<{ rows: {count: number, id: string}[] }>(`
+      select count(e.*), ${idField}
+      from paper p
+      join entity e on e.paper_id = p.s2_id
+      join (
+        select paper_id, max(version) as max_version
+        from entity
+        group by paper_id order by paper_id
```

Comment:
out of curiosity, what is the purpose of ordering by paper_id?

Reply:
In this case, it's a leftover from me figuring out what the queries should be :)